app/models/raw_email.rb

```diff
@@ -95,14 +115,16 @@ def mail!
   end
 
   def data=(d)
-    FileUtils.mkdir_p(directory) unless File.exist?(directory)
-    File.atomic_write(filepath) do |file|
-      file.binmode
-      file.write(d)
-    end
+    source.attach(
+      io: StringIO.new(d.to_s),
+      filename: incoming_message_id,
```

Comment:
Should we take the opportunity to give these an extension (`"#{incoming_message_id}.eml"`)?

Reply:
Changed in 5786d85b4cb0fc42e8a82d5d575240e6e6ef4289

Reply:
👍  I just noticed there's an `active_storage_blobs.filename` column. I guess the intention is to use that attribute if you initiate an app -> client download (rather than storage backend -> app).

Reply:
From my tests these download correctly